### PR TITLE
Minimize build dependencies for 21.3

### PR DESCRIPTION
**Pull request by zakkak**
Removes as many dependencies I was able to remove from truffle and polyglot including the TRUFFLE_ASM jar.

### CODE REVIEW

File: build.java [hunk 1]
```diff
@@ -796,21 +796,99 @@ static void build(
...
-            "truffle:TRUFFLE_API",
-            "com.oracle.svm.truffle.nfi",
-            "com.oracle.svm.truffle.nfi.posix",
-            "com.oracle.svm.truffle.nfi.windows",
-            "extracted-dependency:truffle:LIBFFI_DIST",
-            "extracted-dependency:truffle:TRUFFLE_NFI_NATIVE/include/*",
-            "file:src/com.oracle.svm.libffi/include/svm_libffi.h");
+        Path suitePy = Path.of("substratevm", "mx.substratevm", "suite.py");
+        Path path = mandrelRepo.apply(suitePy);
+        List<String> dependencies = Arrays.asList(
```
Comment (jerboaa) [inline build.java:801]: For clarity call it `removedDeps`, `toRemoveDeps` or some such.

**Review by jerboaa — APPROVED** [review]
Looks fine.

**Review by jerboaa — APPROVED** [review]
LGTM

**Comment by zakkak** [comment]
Being tested in https://github.com/graalvm/mandrel/actions/runs/1602955982

**Pull request merged**